Keep NetApp volume's name when managing it

chuan137 opened the pull request:
We are managing a NetApp volume, which was in Manila db, but currently not. We would like to keep the backend configurations unchanged, including volume name, junction-path, export-policy and etc. With this PR, we can achieve this by providing a driver option, i.e. "--driver-options instance_id=[uuid from volume nanme]". 

NOTE: This PR does not guarantee export policy rules are imported into manila DB. Updating share access rule may overwrite policy rules.

Change-Id: I7fa4788e6448e4f3b46258fa0da937cf2968d259

Carthaca commented on manila/share/drivers/netapp/dataontap/cluster_mode/lib_base.py:
```diff
@@ -2178,20 +2196,25 @@ def _manage_container(self, share, vserver, vserver_client):
-        vserver_client.unmount_volume(volume_name)
-        vserver_client.set_volume_name(volume_name, share_name)
-        vserver_client.mount_volume(share_name)
+        if volume_name != share_name:
```
why are these behind a separate condition and not in the if rename: block above?

Carthaca reviewed (commented):
fyi: some py3 tests are failing because `_manage_container` got a new argument

> TypeError: _manage_container() missing 1 required positional argument: 'options'

chuan137 commented:
```
stnpca1-st005::> vol show -comment *test-manage* -fields policy,node,junction-path
vserver                                 volume                                     policy                                      junction-path                               node           
--------------------------------------- ------------------------------------------ ------------------------------------------- ------------------------------------------- ---------------
ma_988362e3-84d4-4f80-89f2-b80f616eeb1e share_72c85fe1_cab1_4710_90e5_5095522b3027 policy_72c85fe1_cab1_4710_90e5_5095522b3027 /share_72c85fe1_cab1_4710_90e5_5095522b3027 stnpa1-01-st005
```

and after manage

```
MariaDB [manila]> select * from share_instances where share_id = '20573ae5-dee9-42d1-931e-6f14f2af3f2f'\G
*************************** 1. row ***************************
            created_at: 2024-09-19 14:51:06.173686
            updated_at: 2024-09-19 14:51:09.940604
            deleted_at: NULL
               deleted: False
                    id: 72c85fe1-cab1-4710-90e5-5095522b3027
              share_id: 20573ae5-dee9-42d1-931e-6f14f2af3f2f
                  host: manila-share-netapp-stnpca1-st005@stnpca1-st005#aggr_ssd_stnpa1_01_st005_1
                status: available
          scheduled_at: 2024-09-19 14:51:06
           launched_at: 2024-09-19 14:51:09
         terminated_at: NULL
      share_network_id: a0545095-2636-4f2b-95f9-07c50c5fa556
       share_server_id: 988362e3-84d4-4f80-89f2-b80f616eeb1e
  availability_zone_id: 6dc7acdd-7eca-49f6-b266-060c6aca664a
   access_rules_status: active
         replica_state: active
         share_type_id: 6ec58428-82d0-4880-af5d-5ce7459ed209
cast_rules_to_readonly: 0
              progress: NULL
1 row in set (0.000 sec)

MariaDB [manila]> select * from share_instance_export_locations where share_instance_id = '72c85fe1-cab1-4710-90e5-5095522b3027'\G
*************************** 1. row ***************************
               id: 2503429
       created_at: 2024-09-19 14:51:06.209604
       updated_at: 2024-09-19 14:51:09.877352
       deleted_at: NULL
          deleted: 0
             path: 10.180.34.10:/share_72c85fe1_cab1_4710_90e5_5095522b3027
share_instance_id: 72c85fe1-cab1-4710-90e5-5095522b3027
    is_admin_only: 0
             uuid: 7787537f-2f14-4d57-9319-ec2e11ab3e09
*************************** 2. row ***************************
               id: 2503430
       created_at: 2024-09-19 14:51:09.881848
       updated_at: 2024-09-19 14:51:09.877353
       deleted_at: NULL
          deleted: 0
             path: 10.180.34.83:/share_72c85fe1_cab1_4710_90e5_5095522b3027
share_instance_id: 72c85fe1-cab1-4710-90e5-5095522b3027
    is_admin_only: 0
             uuid: b0d98eee-d495-4a1e-82c2-ae51a7c44e6e
2 rows in set (0.000 sec)
```

chuan137 commented:
I am going to test the access rules tomorrow
